src/preferences.js

```diff
@@ -440,6 +457,45 @@ var CPUPowerPreferences = class CPUPowerPreferences {
         dialog.hide();
     }
 
+    onUninstallButtonClicked(button) {
+        let uninstallDialogBuilder = new Gtk.Builder();
+        uninstallDialogBuilder.add_objects_from_file(GLADE_FILE, ["UninstallMessageDialog"]);
+        let dialog = uninstallDialogBuilder.get_object("UninstallMessageDialog");
+        let uninstallButton = uninstallDialogBuilder.get_object("UninstallDialogUninstall");
+        let cancelButton = uninstallDialogBuilder.get_object("UninstallDialogCancel");
+        let parentWindow = this.MainWidget.get_toplevel();
+        dialog.set_transient_for(parentWindow);
+        uninstallButton.connect("clicked", () => {
+            attempt_uninstallation(() => {
+                dialog.close();
+                GLib.spawn_sync(
+	                  null,
+	                  [
+                        'gdbus',
+                        'call',
+                        '--session',
+                        '--dest',
+                        'org.gnome.Shell',
+                        '--object-path',
+                        '/org/gnome/Shell',
+                        '--method',
+                        'org.gnome.Shell.Extensions.ReloadExtension',
+                        'cpupower@mko-sl.de',
+                    ],
```

Comment:
This is a bit hacky, does anyone have a better idea to reload the extension?

Reply:
Maybe `gnome-extensions disable cpupower@mko-sl.de` followed by `gnome-extensions enable cpupower@mko-sl.de` ?

Reply:
Also thought about that... But it's still a command-line call, and does not reload the extension but only restart it. Normally this should not be a problem, but let's better be safe :sweat_smile:

Reply:
On no, I just realized that `ReloadExtension` got deprecated in 3.36, let's use your solution!

Reply:
I think for a full reload a restart of the gnome-session is needed because, for example, if you uninstall the extension it still shows up in `gnome-extensions list` until the gnome-session is restarted ... so apparently it's cashed somehow.

Reply:
Yes, for now a restart of gnome-shell is needed to fully reload an extension. This is only relevant for the developing workflow. For the uninstall action it is sufficient to disable/enable the extension, as the extension code does not change. I don't understand why the reload functionality got deprecated...

Reply:
You could try `ExtensionManager.reloadExtension()`
[Gnome Source](https://gitlab.gnome.org/GNOME/gnome-shell/-/blob/master/js/ui/extensionSystem.js#L339),  [implementation exsample](https://github.com/nls1729/acme-code/blob/master/extension-reloader/extension.js)

Reply:
Let's try it out!

Reply:
@mastercaution do you know how to get the ExtensionManager instance?

Reply:
I tried out a few things. In `extension.js` and everything what it calls (e.g. `indicator.js`) you can use `extensionManager`, because `extension.js` is called by gnome-shell itself I guess.

Reply:
I'm looking whether we can do something similar with `misc.extensionUtils` [(Source)](https://gitlab.gnome.org/GNOME/gnome-shell/-/blob/master/js/misc/extensionUtils.js) because it also makes use of the `extensionManager` in `getCurrentExtension()`

Reply:
Yes, some proper documentation of the gnome-shell API would definitely help :sweat_smile:

Reply:
I tried various things now. `extensionManager.reloadExtension(...)` does not do what we want it to do. Everything seems to do a restart but not a full reload. 
This might not be a problem for the normal use of the extension but it would be very time-saving in development. A restart of the gnome-shell is still required.
Whatever, there seems to be no solution for it.

Reply:
Moved to `gnome-extensions` tool for gnome >=3.34, use the now deprecated `gnome-shell-extension-tool` in older gnome-shell versions.